ksql-common/src/main/java/io/confluent/ksql/function/KsqlFunction.java

```diff
@@ -177,9 +178,11 @@ private void checkMatchingReturnTypes(final Schema s1, final Schema s2) {
     if (!SchemaUtil.areCompatible(s1, s2)) {
       throw new KsqlException(String.format("Return type %s of UDF %s does not match the declared "
                                                 + "return type %s.",
-                                            s1.toString(),
+                                            SchemaConverters.connectToSqlConverter().toSqlType(
```

Comment:
I think it may actually be preferable to use the SqlType#toString instead of converting it to Connect schema first! That way the user gets `ARRAY<INTEGER>` in sql syntax.

Reply:
I am actually converting from Schema to SqlType so that the user gets the message in sql syntax